patches/v8/fix_use_allocationtype_kold_in_v8_scriptormodule_legacy_lifetime.patch

```diff
@@ -0,0 +1,24 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Mon, 22 Nov 2021 16:47:28 +0100
+Subject: fix: use AllocationType::kOld in V8_SCRIPTORMODULE_LEGACY_LIFETIME
+
+See https://chromium-review.googlesource.com/c/v8/v8/+/3211575.
+Enabling V8_SCRIPTORMODULE_LEGACY_LIFETIME is necessary to fix ESM in
+Node.js, but Blink expects old allocations so we change this in the
+interim while Node.js works to remove their usage of WeakRefs to
+ScriptOrModule.
+
+diff --git a/src/heap/factory-base.cc b/src/heap/factory-base.cc
+index 89eb0defc2c1f1d2f6f59c3fb107d844e0c7f986..8a19a38c625daf1f48a5dffc109f73c41f465faa 100644
+--- a/src/heap/factory-base.cc
++++ b/src/heap/factory-base.cc
+@@ -284,7 +284,7 @@ Handle<Script> FactoryBase<Impl>::NewScriptWithId(
+ 
+ template <typename Impl>
+ Handle<ArrayList> FactoryBase<Impl>::NewArrayList(int size) {
+-  Handle<FixedArray> fixed_array = NewFixedArray(size + ArrayList::kFirstIndex);
++  Handle<FixedArray> fixed_array = NewFixedArray(size + ArrayList::kFirstIndex, AllocationType::kOld);
```

Comment:
I would suggest to report the crash with chromium/v8 using a minimal sample with content_shell or d8 and disable the relevant test on our side with link to the crbug.

Reply:
we can try - but my concern is that isn't one or two
tests, it's ESM functionally entirely. Do we feel ok indefinitely breaking that until this is resolved upstream?

Reply:
or do you mean disabling the webSecurity test? i can do that but it also causes the same crash in a PDF test and it's less clear how that's happening since webSecurity isn't involved there

Reply:
Yup to clarify, I am good with the flag `v8_scriptormodule_legacy_lifetime` to maintain compatibility with node esm behavior. What I am suggesting is to either alter the `webSecurity` test case or disable it until the upstream issue is resolved instead of v8 patch that changes object allocation space for all scenarios. Also another question, does the crash happen when loading any scripts on a renderer with `webSecurity: false` or is it only the test case that you have posted in the upgrades channel ?

>  but it also causes the same crash in the PDF test that loads a large html file

One thing that stands out is that even this test uses `https://github.com` source in `print-to-pdf.html` so maybe this crash is localized to certain websites, atleast known is `https://github.com`. I had a similar localized crash with this site in the past upgrades, and I temporarily disabled the spec while the related upstream issue was resolved ex: https://bugs.chromium.org/p/chromium/issues/detail?id=1085667

Reply:
@deepak1556 ah, ok! and the crash happens when loading a url as a `src` for a script in the renderer but not when executing other code inside the script tag that i've been able to determine.

Reply:
should be set!

Reply:
ah yeah, on it!

Reply:
done!